diff --git a/source/darkplaces/makefile.inc b/source/darkplaces/makefile.inc
index 9c66ed1..b48f3cc 100644
--- a/source/darkplaces/makefile.inc
+++ b/source/darkplaces/makefile.inc
@@ -202,7 +202,7 @@ DO_CC=$(CC) $(CFLAGS) -c $< -o $@
 # Link
 LDFLAGS_DEBUG=-g -ggdb $(OPTIM_DEBUG) -DSVNREVISION=`{ test -d .svn && svnversion; } || { test -d .git && git describe --always; } || echo -` -DBUILDTYPE=debug
 LDFLAGS_PROFILE=-g -pg -fprofile-arcs $(OPTIM_RELEASE) -DSVNREVISION=`{ test -d .svn && svnversion; } || { test -d .git && git describe --always; } || echo -` -DBUILDTYPE=profile
-LDFLAGS_RELEASE=$(OPTIM_RELEASE) -DSVNREVISION=`{ test -d .svn && svnversion; } || { test -d .git && git describe --always; } || echo -` -DBUILDTYPE=release
+LDFLAGS_RELEASE?=$(OPTIM_RELEASE) -DSVNREVISION=`{ test -d .svn && svnversion; } || { test -d .git && git describe --always; } || echo -` -DBUILDTYPE=release
 
 
 ##### UNIX specific variables #####
